### PR TITLE
feat: Enable reorder blocks

### LINKED ISSUE #266: Reorder blocks
Allow reordering by swapping adjacent blocks.

### PR DESCRIPTION
**Pull request by mogavin**
* Enable reorder blocks via `movableBlocks` prop;
* Adjust block spacing

Closes #266

### CODE REVIEW

File: src/styles/move-control.scss [hunk 1]
```diff
@@ -0,0 +1,67 @@
+.move-control {
```
Comment (Chavao) [inline src/styles/move-control.scss]: Hi @mogavin, Could you please put the copyright holder and license inside this file?

File: src/swapDataBlock.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import { EditorState } from "draft-js";
```
Comment (Chavao) [inline src/swapDataBlock.js]: Hi @mogavin, Could you please put the copyright holder and license inside this file?

File: src/components/MoveControl.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import React from "react";
```
Comment (Chavao) [inline src/components/MoveControl.js]: Hi @mogavin, Could you please put the copyright holder and license inside this file?

File: src/swapDataBlock.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+const swapBlocks = ({ currentKey, targetKey, editorState }) => {
+    const contentState = editorState.getCurrentContent(),
+      selectionState = editorState.getSelection(),
+      blockMap = contentState.getBlockMap(),
+      keySeq = blockMap.keySeq(),
+      getIndex = key => keySeq.findIndex(k => k === key),
+      insertBlock = (target, key, val) => {
```
Comment (Chavao) [inline src/swapDataBlock.js]: Could you initialize the variables in each line, to keep standardized with the current codebase?

File: src/swapDataBlock.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      newContentState = contentState.merge({
+        blockMap: newBlocks,
+        selectionAfter: selectionState.merge({
+          hasFocus: true
+        })
+      });
+
+    return EditorState.push(editorState, newContentState, "move-block");
+  },
+  swap = ({ getTargetKeyFn, editorState, currentKey }) => {
```
Comment (Chavao) [inline src/swapDataBlock.js]: Could you initialize the variables in each line, to keep standardized with the current codebase?

File: src/components/MoveControl.js (hunk 3 above)
Comment (mogavin) [reply src/components/MoveControl.js]: Done !

File: src/styles/move-control.scss (hunk 1 above)
Comment (mogavin) [reply src/styles/move-control.scss]: Done !

File: src/swapDataBlock.js (hunk 2 above)
Comment (mogavin) [reply src/swapDataBlock.js]: Done !

File: src/swapDataBlock.js (hunk 5 above)
Comment (mogavin) [reply src/swapDataBlock.js]: Done !

File: src/components/MoveControl.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    e.stopPropagation();
+  };
+  return (
+    <div className="options">
+      <button
+        data-testid={`swap-up-${id}`}
+        className="options__button options__button--up"
+        onClick={onClickUp}
+        disabled={disableUp}
+        {...{ onPointerDown }}
```
Comment (icaromh) [inline src/components/MoveControl.js]: I think that's better to be more explicit here
```js
{...{ onPointerDown }}
```
passing the prop itself `onPointerDown={onPointerDown}`

File: src/components/MoveControl.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      >
+        <icons.DropdownArrow />
+      </button>
+      <button
+        data-testid={`swap-down-${id}`}
+        className="options__button"
+        onClick={onClickDown}
+        disabled={disableDown}
+        {...{ onPointerDown }}
```
Comment (icaromh) [inline src/components/MoveControl.js]: the same here

File: src/components/MoveControl.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  );
+};
+
+const Control = ({ children, id, onClickUp, onClickDown, isFirst, isLast }) => (
+  <div className="move-control" id={`move-control-${id}`}>
+    <div className="move-control__target" data-testid={`block-${id}`}>
+      {children}
+    </div>
+    <Options
+      {...{ id, onClickUp, onClickDown }}
```
Comment (icaromh) [inline src/components/MoveControl.js]: Here looks great :D

File: src/components/MoveControl.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+export default ({
+  wrapper,
+  swapUp,
+  swapDown,
+  children,
+  isFirstBlock,
+  isLastBlock
+}) => {
+  const arrayChildren = React.Children.toArray(children),
+    firstChildKey = arrayChildren[0].props.children.key,
```
Comment (icaromh) [inline src/components/MoveControl.js]: Same what @Chavao already said 😄 
> Could you initialize the variables in each line, to keep standardized with the current codebase?

File: src/components/MoveControl.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    lastChildKey = arrayChildren[arrayChildren.length - 1].props.children.key,
+    Controlled = ({ keySwapUp, keySwapDown, children }) => {
+      const onClickUp = () => swapUp(keySwapUp),
```
Comment (icaromh) [inline src/components/MoveControl.js]: here too

File: src/components/MoveControl.js (hunk 6 above)
Comment (mogavin) [reply src/components/MoveControl.js]: Done!

File: src/components/MoveControl.js (hunk 7 above)
Comment (mogavin) [reply src/components/MoveControl.js]: Done!

File: src/components/MoveControl.js (hunk 9 above)
Comment (mogavin) [reply src/components/MoveControl.js]: Done!

File: src/components/MoveControl.js (hunk 10 above)
Comment (mogavin) [reply src/components/MoveControl.js]: Done!

**Pull request merged**